src/eos-updater-util.c

```diff
@@ -352,6 +366,16 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
   shuffle_out_values (upgrade_remote, o_remote, NULL);
   shuffle_out_values (upgrade_ref, p_ref, NULL);
 
-  out:
-    return ret;
+out:
+  if ((p_ref || on_hold) && !metric_sent)
+    {
+      emtr_event_recorder_record_event (emtr_event_recorder_get_default (),
+                                        EOS_UPDATER_BRANCH_SELECTED,
+                                        g_variant_new ("(sssb)", vendor,
+                                                       product,
+                                                       on_hold ? o_ref : p_ref,
+                                                       on_hold));
+      metric_sent = TRUE;
+    }
```

Comment:
I kinda feel like this is the wrong place to be shoving this, but that would require providing the vendor and product back out to the caller. That would require more refactoring. For another day...